### ISSUE TITLE
POD cannot communicate to itself via k8s-svc 

### ISSUE THREAD

**Issue by slawekgh** [openservicemesh/osm#4340]
I have k8s deployment with 4 PODS and 1 corresponding k8s-SVC 

when I exec into one of these 4 PODS and start curling on k8s-SVC I have 3 correct answers (from 3 other PODS) but 4th session goes to this POD that I'm on that 

and I got the message: 
```
upstream connect error or disconnect/reset before headers. reset reason: connection failure, transport failure reason: TLS error: 268435703:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER
```
when I scale deployment to 2 PODS I have 1 correct answer and 1 with error 

and analogously when I scale to 6 PODS I have 5 correct answers but when it hit that POD that I'm on that I got once per 6 error message 
```
upstream connect error or disconnect/reset before headers. reset reason: connection failure, transport failure reason: TLS error: 268435703:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER
```

seems like POD cannot communicate to itself via k8s-svc 

tested on AKS with OSM 0.9.2 and 0.11

**Comment by shashankram** [comment]
@slawekgh We do not support a pod communicating to itself via a k8s svc, and instead such a request must be directed to localhost so that it bypasses the sidecar proxy.

**Comment by shashankram** [comment]
@slawekgh I have merged a change that should address this issue. Kindly let me know if it works for your scenario. There is a specific scenario where if the server initiates the first byte of data instead of a client, the request will delay for 15s in Envoy before it gets through. To avoid this, you could set the `appProtocol` on `service.spec.ports[]` to `tcp-server-first`.

**Comment by slawekgh** [comment]
@shashankram thanks a lot , how to install that ? 
I think it is not available as AKS - OSM add-on , so only manual installation ?

**Comment by shashankram** [comment]
> @shashankram thanks a lot , how to install that ? I think it is not available as AKS - OSM add-on , so only manual installation ?

This change has been merged to main, and should be available in the next release (v1.0). For testing, I recommend creating a test cluster and installing the control plane from the `main` branch of this repo. Refer to https://github.com/openservicemesh/osm/tree/main/docs/development_guide on how you can compile the code from source and build the images.

**Comment by shashankram** [comment]
Hey @slawekgh, would it be possible to test the fix using [v1.0.0-rc.4](https://github.com/openservicemesh/osm/releases/tag/v1.0.0-rc.4)?
